src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractPackage`2.cs

```diff
@@ -95,9 +95,6 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
                 Workspace.AdviseSolutionEvents(solution);
             }
 
-            // Ensure services that must be created on the UI thread have been.
-            HACK_AbstractCreateServicesOnUiThread.CreateServicesOnUIThread(ComponentModel, RoslynLanguageName);
```

Comment:
The argument here is since we're still doing this on subject buffer connection, it's fine?

Reply:
yes, because if there is a code file open, it will hit the subject buffer connection first, before the solution load completes